common/models/user.js

```diff
@@ -668,6 +668,45 @@ module.exports = function(User) {
       next();
     });
 
+    // Delete old sessions once email is updated
+    var oldEmail, newEmail, oldId, newId;
+    UserModel.observe('before save', function(ctx, next) {
+      if (ctx.instance) {
+        if (ctx.instance.id !== undefined) {
+          ctx.hookState.id = ctx.instance.id;
+          oldId = ctx.hookState.id;
+        }
+      }
+      if (ctx.data) {
+        ctx.hookState.id = ctx.where.id;
```

Comment:
`ctx.where.id` may not be always present, it also may not be a simple value like `{ where: { id: 123 }}`, but a condition like `{ where: { id: { in: [123, 456]}}}`.

This typically happens when calling `updateAll`, i.e. performing a bulk update.